audio/out/ao_pipewire.c

```diff
@@ -580,24 +596,41 @@ static int init(struct ao *ao)
 
     pw_properties_setf(props, PW_KEY_NODE_RATE, "1/%d", ao->samplerate);
 
-    enum spa_audio_format spa_format = af_fmt_to_pw(ao, ao->format);
-    if (spa_format == SPA_AUDIO_FORMAT_UNKNOWN) {
-        ao->format = AF_FORMAT_FLOATP;
-        spa_format = SPA_AUDIO_FORMAT_F32P;
-    }
+    if (af_fmt_is_spdif(ao->format)) {
+        enum spa_audio_iec958_codec spa_codec = af_fmt_to_codec(ao, ao->format);
+        if (spa_codec == SPA_AUDIO_IEC958_CODEC_UNKNOWN) {
+            ao->format = AF_FORMAT_FLOATP;
+            spa_codec = SPA_AUDIO_IEC958_CODEC_PCM;
+        }
```

Comment:
Just to be explicit, if the audio stream is actually PCM, it will not not hit this path because `af_fmt_is_spdif` won't return true, right?

In that case, should we actually be sending the unknown samples at all? Could this lead to garbage actually going out and damaging hardware and/or people's ears?

Reply:
Makes sense, I changed it.

But don't we have the same issue on `SPA_AUDIO_FORMAT_UNKNOWN` in the PCM codepath?

Reply:
I suppose so? We didn't think too much about it at the time, but it's better to error out if we end up with UNKNOWN.  If anyone has a legitimate case, they'll tell us about it, presumably. Better than damaging something/someone.

Reply:
Sounds good, I added a commit to this PR for it.